cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/AbstractLogHandler.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.gateway.handlers;
+
+import co.cask.cdap.api.dataset.lib.CloseableIterator;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.logging.LoggingContext;
+import co.cask.cdap.internal.app.store.RunRecordMeta;
+import co.cask.cdap.logging.LoggingConfiguration;
+import co.cask.cdap.logging.filter.Filter;
+import co.cask.cdap.logging.filter.FilterParser;
+import co.cask.cdap.logging.gateway.handlers.store.ProgramStore;
+import co.cask.cdap.logging.read.Callback;
+import co.cask.cdap.logging.read.LogEvent;
+import co.cask.cdap.logging.read.LogOffset;
+import co.cask.cdap.logging.read.LogReader;
+import co.cask.cdap.logging.read.ReadRange;
+import co.cask.http.AbstractHttpHandler;
+import co.cask.http.HttpResponder;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Abstract Class that contains commonly used methods for log Http Requests.
+ */
+public class AbstractLogHandler extends AbstractHttpHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractLogHandler.class);
+
+  private final LogReader logReader;
+  private final String logPattern;
+
+  public AbstractLogHandler(LogReader logReader, CConfiguration cConfig) {
+    this.logReader = logReader;
+    this.logPattern = cConfig.get(LoggingConfiguration.LOG_PATTERN, LoggingConfiguration.DEFAULT_LOG_PATTERN);
+  }
+
+  protected void doGetLogs(HttpResponder responder, LoggingContext loggingContext,
```

Comment:
I do not think this is possible since this method is called by `LogHandler` by three different endpoints to get logs with or without runid, get logs for system component. And the endpoints are like `/namespaces/{namespace-id}/apps/{app-id}/{program-type}/{program-id}/runs/{run-id}/logs`, where in `PreviewHttpHandler`, the endpoint is like `/namespaces/{namespace-id}/previews/{preview-id}/logs`.